watertap/core/zero_order_costing.py

```diff
@@ -965,6 +966,177 @@ def cost_mabr(blk):
+                * mixer_cost,
+                to_units=blk.config.flowsheet_costing_block.base_currency,
+            )
+        )
+        blk.eq_DCC_bead = pyo.Constraint(
+            expr=blk.DCC_bead
+            == pyo.units.convert(
+                blk.unit_model.volume * bead_bulk_density * bead_cost,
+                to_units=blk.config.flowsheet_costing_block.base_currency,
+            )
+        )
+        blk.eq_DCC_membrane = pyo.Constraint(
+            expr=blk.DCC_membrane
+            == pyo.units.convert(
+                blk.unit_model.get_inlet_flow(0)
+                * membrane_sidestream_fraction
+                * membrane_specific_size
+                * membrane_cost,
+                to_units=blk.config.flowsheet_costing_block.base_currency,
+            )
+        )
+        blk.eq_DCC_vacuum = pyo.Constraint(
+            expr=blk.DCC_vacuum
+            == pyo.units.convert(
+                blk.unit_model.properties_byproduct[0].flow_mass_comp[
+                    blk.unit_model._gas_comp
+                ]
+                * vacuum_cost,
+                to_units=blk.config.flowsheet_costing_block.base_currency,
+            )
+        )
+
+        expr = (
+            blk.DCC_reactor
+            + blk.DCC_mixer
+            + blk.DCC_bead
+            + blk.DCC_membrane
+            + blk.DCC_vacuum
+        )
+
+        # Determine if a costing factor is required
+        factor = parameter_dict["capital_cost"]["cost_factor"]
+
+        if factor == "TPEC":
+            expr *= blk.config.flowsheet_costing_block.TPEC
+        elif factor == "TIC":
+            expr *= blk.config.flowsheet_costing_block.TIC
+
+        blk.capital_cost_constraint = pyo.Constraint(expr=blk.capital_cost == expr)
+
+        # Add fixed operating cost variable and constraint
+        blk.fixed_operating_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency
+            / blk.config.flowsheet_costing_block.base_period,
+            bounds=(0, None),
+            doc="Fixed operating cost of unit",
+        )
+        blk.fixed_operating_cost_constraint = pyo.Constraint(
+            expr=blk.fixed_operating_cost
+            == pyo.units.convert(
+                blk.DCC_bead * bead_replacement_factor,
+                to_units=blk.config.flowsheet_costing_block.base_currency
+                / blk.config.flowsheet_costing_block.base_period,
+            )
+        )
+
+        # Register operating cost flows
+        blk.config.flowsheet_costing_block.cost_flow(
+            blk.unit_model.electricity[t0], "electricity"
+        )
+        blk.config.flowsheet_costing_block.cost_flow(blk.unit_model.heat[t0], "heat")
+        blk.config.flowsheet_costing_block.cost_flow(
+            blk.unit_model.properties_byproduct[0].flow_mass_comp[
+                blk.unit_model._gas_comp
+            ],
+            blk.unit_model._gas_comp + "_product",
```

Comment:
First, see my comment in the METAB unit model. Secondly, this could give rise to unexpected behaviours if `blk.unit_model._gas_comp` were not a string for some reason (and we have no guards in place to prevent that).

Finally, I am a little hesitant with the current setup as this means our list of costed components is now dynamic. I supposed this just required the end-user to declare any new product component they wish to add via a METAB reactor, but that probably needs to be clearly documented (and runs against what we have generally done in the WT3 models, where everything was concretely defined).